server/game/GameActions/PutIntoPlayAction.js

```diff
@@ -39,7 +39,10 @@ class PutIntoPlayAction extends CardGameAction {
             return;
         }
 
-        if (card.anyEffect('entersPlayUnderOpponentsControl') && card.owner.opponent) {
+        if (
```

Comment:
Take a look at Card.js#setupKeywordAbilities

I believe treachery can be implemented as:

```
this.abilities.keywordPersistentEffects.push(
            this.persistentEffect({
                printedAbility: false,
                condition: () => this.hasKeyword('treachery'),
                match: this,
                effect: AbilityDsl.effects.entersPlayUnderOpponentsControl()
            })
        );
```

Reply:
Good idea, thanks.  Done.